### PR TITLE
refactor mkdir into a common function

### PR DESCRIPTION
**Pull request by Fredy-Z**
#656 
Create `test-infra/shared/common/common.go` to hold the `CreateDir` function, and use it to replace the duplicate code.

We can continuously add common functions used by multiple components in this file.

### CODE REVIEW

File: shared/loadgenerator/loadgenerator.go [hunk 1]
```diff
@@ -109,10 +110,8 @@ func (g *GeneratorOptions) RunLoadTest(resolvableDomain bool) (*GeneratorResults
 // SaveJSON saves the results as Json in the artifacts directory
 func (gr *GeneratorResults) SaveJSON(testName string) error {
 	dir := prow.GetLocalArtifactsDir()
```
Comment (adrcunha) [inline shared/loadgenerator/loadgenerator.go]: Is it worth making GetLocalArtifactsDir() creating the dir automatically, if it doesn't exist? Or maybe encapsulate this pattern in a "GetOrCreate" function?

Comment (Fredy-Z) [reply shared/loadgenerator/loadgenerator.go]: Seems it's not worth changing this function, [here](https://github.com/knative/test-infra/blob/4ce16d390c55c71290f3b285bdfe37a1c5490304/tools/flaky-test-reporter/result.go#L118) it's only used for retrieving the path string. And several test cases are also using it for such purpose.

**Pull request merged**